### DISCUSSION TITLE
Get Half Data through StartNotification() on Android 14 (API Level 34).

### DISCUSSION THREAD

**Discussion by GovindMobilesutra** [capacitor-community/bluetooth-le#682] [Q&A]
When I scan and connect to the device and initiate communication, my messages are successfully sent to the device. However, when the device responds with a large data payload (such as a JSON object with 50 key-value pairs), I only receive the last portion of the data that the device sends.

**Comment by peitschie** [comment]
Hi @GovindMobilesutra 

BLE has a maximum notification size that can vary between iOS and Android, and between different phones even... but it's usually in the range of 20 bytes up to only about 512 bytes.

If the payload you're trying to send via the notification exceeds this... it depends on what the underlying peripheral library does as to what your application will see.

To send larger payloads, you'll potentially need to use Indications rather than Notifications (see [this link](https://www.ezurio.com/support/faqs/what-difference-between-indication-and-notification) for a short discussion on the differences), and likely do some reassembly of the fragmented payload in your application.

Alternatively, if you can find a way to reduce the notification size to be below the minimum MTU size the phones you're targeting supports... this is a neater and often simpler approach.

**Reply by GovindMobilesutra** [reply]
I am concatenating the response which the peripheral is sending which is working OK till Android 13, but on Android 14 I am only getting the last chunk

**Reply by peitschie** [reply]
I wonder if there's any connection to #688 

Sorry, I don't have any clear insight as to what's going on here. Android 14 (with the latest version of this plugin) leverages a new [onCharacteristicChanged](https://developer.android.com/reference/android/bluetooth/BluetoothGattCallback#onCharacteristicChanged(android.bluetooth.BluetoothGatt,%20android.bluetooth.BluetoothGattCharacteristic,%20byte[])) API, which is only on Android 14+.

See https://github.com/capacitor-community/bluetooth-le/blob/ee6d9671090eb1bca6841086e8da95cde074fd54/android/src/main/java/com/capacitorjs/community/plugins/bluetoothle/Device.kt#L208-L234

Perhaps you've hit an inconsistency Android has introduced here?
